core/thread/thread_variable_set_spec.rb

```diff
@@ -10,17 +10,50 @@
   end
 
   it "returns the value set" do
-    (@t.thread_variable_set :a, 2).should == 2
+    @t.thread_variable_set(:a, 2).should == 2
   end
 
   it "sets a value that will be returned by #thread_variable_get" do
-    @t.thread_variable_set :a, 49
+    @t.thread_variable_set(:a, 49)
     @t.thread_variable_get(:a).should == 49
   end
 
   it "sets a value private to self" do
-    @t.thread_variable_set :thread_variable_get_spec, 82
+    @t.thread_variable_set(:thread_variable_get_spec, 82)
     @t.thread_variable_get(:thread_variable_get_spec).should == 82
     Thread.current.thread_variable_get(:thread_variable_get_spec).should be_nil
   end
+
+  it "converts a String key into a Symbol" do
+    @t.thread_variable_set('a', 49)
+    @t.thread_variable_get(:a).should == 49
+  end
```

Comment:
**suggestion:** the test title is not accurate enough as for me.

The test doesn't check that String is converted into Symbol (e.g. it could expect String#to_sym method to be called). It just checks that Symbol and String keys could be used interchangeable.

There is a similar issue with changes in the `thread_variable_spec.rb` file.